File: 250323/완전수/perfect-number.cpp
#include <iostream>
using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int start = 0, end = 0, count = 0;
    cin >> start >> end;
    for(int i = start; i <= end; i++) {
        int sum = 0;
        for(int j = 1; j < i; j++) {
            if(i % j == 0) {
                sum += j;
            }
        }
        if(sum == i) {
            count++;
        }
    }
    cout << count;
    return 0;
}

File: 250323/완전수/perfect-number.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int start = sc.nextInt();
        int end = sc.nextInt();
        int count = 0;
        for(int i = start; i <= end; i++) {
            int sum = 0;
            for(int j = 1; j <= i; j++) {
                if(i % j == 0 && j != i) {
                    sum += j;
                }
            }
            if(sum == i) {
                count++;
            }
        }
        System.out.println(count);
        sc.close();
    }
}